[Bug] cache restore tags undefined

## Describe the bug

on cache restore, there is an intermittent undefined value for tags

## To Reproduce

Steps to reproduce the behavior:

?

## Expected behavior

the json object should be stored and loaded without error

## Additional context

```
[ERROR] EXT IS NULL - CONFIGURATION SKIPPED
[DEBUG] CACHE UNSUCCESSFULLY RESTORED
[DEBUG] this.tags(...) is undefined
```

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -307,7 +307,7 @@ export class Ext extends Ecs.System<ExtEvent> {
             auto_tiler: this.auto_tiler,
             entities: this.entities_,
             storages: this.storages,
-            tags: this.tags_,
+            tags: this.tags_.map(set => [...set]),
             free_slots: this.free_slots,
             displays: [
                 this.displays[0],
@@ -382,7 +382,8 @@ export class Ext extends Ecs.System<ExtEvent> {
                 )
             );
             ext.tags_ = data.tags.map(
-                (obj: any) => new Set(Object.entries(obj))
+                (val: any) =>
+                    new Set(Array.isArray(val) ? val : Object.entries(val))
             );
             ext.free_slots = data.free_slots;
             if (data.displays) {

--- a/src/forest.ts
+++ b/src/forest.ts
@@ -89,7 +89,7 @@ export class Forest extends Ecs.World {
             string_reps: this.string_reps,
             entities: this.entities_,
             storages: this.storages,
-            tags: this.tags_,
+            tags: this.tags_.map(set => [...set]),
             free_slots: this.free_slots,
         };
     }
@@ -114,7 +114,8 @@ export class Forest extends Ecs.World {
             Ecs.Storage.fromJSON(obj.store, val => val)
         );
         forest.tags_ = data.tags.map(
-            (obj: any) => new Set(Object.entries(obj))
+            (val: any) =>
+                new Set(Array.isArray(val) ? val : Object.entries(val))
         );
         forest.free_slots = data.free_slots;
         return forest;

